### PR TITLE
Implement file browsing

### PR DESCRIPTION
**Pull request by OshanIvantha**
## Changes made

- Implement the navigation within directories
- Add back navigation
- Implement file loading on page reload

### CODE REVIEW

File: apps/files/src/components/Files-App.vue [hunk 1]
```diff
@@ -1,257 +1,339 @@
...
+                    pathSplit = pathSplit.slice(1, pathSplit.length - 1);
+                    this.path = pathSplit
+                }
+
+                // List all files
+                OC.$client.files.list(absolutePath).then(files => {
+                    // Remove the root element
+                    files = files.splice(1);
+
+                    let delay = Math.random() * (1000 - 200) + 200;
```
Comment (felixheidecke) [inline apps/files/src/components/Files-App.vue]: I wrote this to demonstrate a delay + the loading animation. We can gladly get rid of that :-)

File: apps/files/src/components/Files-App.vue [hunk 2]
```diff
@@ -1,257 +1,339 @@
...
+                            });
+                        });
+
+                        this.self = files.self;
+
+                        this.loading = false;
+
+                        this.resetSelect();
+                    }, delay)
+                }).catch(error => {``
```
Comment (felixheidecke) [inline apps/files/src/components/Files-App.vue]: Whats up with the ticks?

File: apps/files/src/components/Files-App.vue [hunk 3]
```diff
@@ -1,257 +1,339 @@
...
+            },
+            typeOfFile(showHidden) {
+                showHidden = (typeof showHidden !== 'undefined') ? showHidden : false;
+                return _.filter(this.files, 'extension')
+            }
+        }
+    }
 </script>
 
 <style lang="less">
```
Comment (felixheidecke) [inline apps/files/src/components/Files-App.vue]: I'd recommend against "larger" styling inside the vue files and rather have that move to a separate Less file, but we can do this in the future.

File: apps/files/src/components/Files-App.vue (hunk 2 above)
Comment (OshanIvantha) [reply apps/files/src/components/Files-App.vue]: Oh typo I guess.

File: apps/files/src/components/Files-App.vue (hunk 3 above)
Comment (OshanIvantha) [reply apps/files/src/components/Files-App.vue]: They were there initially. I will start separating them.

**Pull request merged**